ccmlib/node.py

```diff
@@ -745,30 +745,59 @@ def stop(self, wait=True, wait_other_notice=False, other_nodes=None, gently=True
         else:
             return False
 
-    def wait_for_compactions(self, idle_timeout=300):
+    def _get_pending_tasks(self, stats, keyspace, column_family):
+        assert keyspace is not None, 'keyspace is not specified'
+        assert column_family is not None, 'column_family is not specified'
+        # "nodetool compactionstats" prints the compaction stats like:
+        # pending tasks: 42
+        # - ks1.cf1: 13
+        # - ks1.cf2: 19
+        # - ks2.cf1: 10
+        pattern = re.compile(r'\-\s+([^:]+):\s*(\d+)')
+        for line in stats.strip().split('\n')[1:]:
+            ks_cf, num = pattern.search(line).groups()
+            if ks_cf == f'{keyspace}.{column_family}':
+                return int(num)
+        return 0
+
+    def wait_for_compactions(self,
+                             keyspace=None,
+                             column_family=None,
+                             idle_timeout=300):
```

Comment:
new optional params should be added as last since existing call sites may not specify the idle_timeput by name and just invoke `node.wait_for_compactions(120)`

Reply:
i agree you point in general. but because we always put keyspace and column_family as the first two argument, i took extra efforts to check all the callers in scylla-dtest and in scylla-ccm, none of the existing callers calls with positional argument.